### DISCUSSION TITLE
How retrieve the collision point's position

### DISCUSSION THREAD

**Discussion by 2016Leo123** [bepu/bepuphysics1#45] [Q&A]
Say, I have a box at 0, 0, 0. another box run and hit it
Space.Add(toAdd);
toAdd.CollisionInformation.Events.DetectingInitialCollision += DetectingInitialCollisionCallback;
public void DetectingInitialCollisionCallback(EntityCollidable sender, Collidable other, CollidablePairHandler collisionPair)
{
    //do something
    Is there anyway retrieve the collision point directly.
    collisionPair.Contacts is an array, go through all its items, collisionPair.Contacts[ii].Contact.Position seems is not correct which I 
    wanna.
}
Thanks Advance

**Comment by RossNordby** [comment]
The pair's contacts should contain the relevant contacts. You could also use the contact-specific events like `CreatingContact` which pass a specific contact, but they'll cover the same contacts as are visible in the pair's contact list.

(The reason why the initial collision event doesn't provide a specific contact as a parameter is that it fires upon the transition from zero to nonzero contacts. Multiple contacts may appear simultaneously, and no contact is the 'first' because they all occurred at the same instant.)
